src/StationAPIConnector.php

```diff
@@ -219,79 +219,47 @@ public function apiQueryPayload($delta = FALSE) {
    * @throws \Drupal\Core\Entity\EntityStorageException
    */
   public function getFormattedStationResponse(array $stations) {
-    $speciality_keys = [
-      'Crime_Investigate_Unit_CIU',
-      'Crime_Prevent_Officer_CPO',
-      'FVIU',
-      'FVLO',
-      'FCLO',
-      'Justice_Of_The_Peace',
-      'LGBTIQ_Liaison_Officer_LLO',
-      'Neighbourhood_Watch',
-      'Pro_active_Police_Unit_PPU',
-      'Regional_Firearms_Officers',
-      'SOCIT',
-      'Victim_Assist_Supp_Officer',
-      'Youth_Resource_Officer',
-    ];
+    $final_stations = $states = $speciality_keys = $speciality_terms_data = [];
 
-    $final_stations = $states = $speciality_terms = [];
-
-    // We want the accessibility terms to be in a certain order.
-    // So initialising it.
-    $accessibility_terms = [
-      'Front entrance ground level' => 'Front entrance ground level',
-      'Automatic entrance door' => 'Automatic entrance door',
-      'Wheelchair/ramp access Internal' => 'Wheelchair/ramp access Internal',
-      'Wheelchair/ramp access External' => 'Wheelchair/ramp access External',
-      'Accessible parking' => 'Accessible parking',
-      'Internal parking lift access' => 'Internal parking lift access',
-      'Automatic lift door' => 'Automatic lift door',
-      'Accessible toilet public/interior' => 'Accessible toilet public/interior',
-      'Hearing loop' => 'Hearing loop',
-      'Audible level recognition' => 'Audible level recognition',
-      'SCOPE Communication Access accredited police station' => 'SCOPE Communication Access accredited police station',
-      'Lift braille buttons' => 'Lift braille buttons',
-      'External/entrance braille signage' => 'External/entrance braille signage',
-      'Internal braille signage' => 'Internal braille signage',
-    ];
+    $speciality_terms = \Drupal::entityTypeManager()
+      ->getStorage('taxonomy_term')
+      ->loadByProperties(['vid' => 'specialty_services_or_facilities']);
 
     $i = 0;
     // Process the data so that we can save it in a format we need.
     foreach ($stations as $key => $station) {
       // Loop through attributes.
       foreach ($station['attributes'] as $attribute) {
-        // Add attribute name as the key.
-        $station[$attribute['name']] = html_entity_decode($attribute['value']);
-
-        if ($attribute['name'] == 'State_Name') {
-          $states[$key] = $attribute['value'];
+        if (html_entity_decode($attribute['value']) == 'Y') {
```

Comment:
@g2410g we should also check if this key is in `$speciality_terms` to future proof this.

Reply:
@gargsuchi Updated the code.